src/hooks/useBalanceChartData/useBalanceChartData.ts

```diff
@@ -389,37 +381,16 @@ export const useBalanceChartData: UseBalanceChartData = args => {
   const cryptoPriceHistoryData = useAppSelector(state =>
     selectCryptoPriceHistoryTimeframe(state, timeframe),
   )
-  const cryptoPriceHistoryDataLoading = useAppSelector(state =>
-    selectPriceHistoriesLoadingByAssetTimeframe(state, assetIds, timeframe),
-  )
 
   const fiatPriceHistoryData = useAppSelector(state =>
     selectFiatPriceHistoryTimeframe(state, timeframe),
   )
-  const fiatPriceHistoryDataLoading = useAppSelector(state =>
-    selectFiatPriceHistoriesLoadingByTimeframe(state, timeframe),
-  )
-
   // loading state
   useEffect(() => setBalanceChartDataLoading(true), [setBalanceChartDataLoading, timeframe])
 
   // calculation
   useEffect(() => {
-    // data prep
-    const hasNoDeviceId = isNil(walletInfo?.deviceId)
```

Comment:
when i initially wrote this, i thought the computation would be the bottleneck, it's not.